### DISCUSSION TITLE
Embedded dashboard: hide edit chart title link

### DISCUSSION THREAD

**Discussion by frlm** [apache/superset#25436] [Q&A / Help]
I am opening this discussion because I am unable to hide the **_edit link_** of the chart within an embedded dashboard, i.e. **_href_** associated with the chart title, users accessing the dashboard should not view it because they do not have permissions to edit the chart. Even if the user is blocked from editing and therefore will not access the displayed link, I would prefer that the user does not see it. 

![image](https://github.com/apache/superset/assets/95374244/ce10d731-6034-4eb3-81a7-170ab5e10e04)

**Comment by Ralkion** [comment]
Hi there.

I'm not a contributor, but I've found that the link you describe on the dashboard is controlled by the permission "Can explore on Superset".

That permission does seem to control some other items as well though, so you may still need it depending on your use-case.

**Comment by MaximDown** [comment]
Hide using CSS

**Comment by Aponte145** [comment]
Hello,

To hide the title of a chart in a dashboard using CSS, you can use the following code in the dashboard

[data-test-chart-id] > div:first-child {
    display: none;
}

If you want to show the title but prevent it from being a clickable link, use this code instead:

[data-test-chart-id] > div:first-child a {
    pointer-events: none;
    cursor: default;
    color: inherit;
    text-decoration: none;
}

I hope this helps!

**Reply by arkpatrick** [reply]
This works very nicely. To hide the "Click to edit Top fields. Use ⌘ + click to open in a new tab." tooltip without hiding the title, I went with the following

.ant-tooltip-inner {
  display: none;
}

**Reply by tabgruedksr** [reply]
It still showed up for me, so I edited your suggestion to this and it worked.
`.ant-tooltip {
  display: none !important;
  opacity: 0 !important;
  visibility: hidden !important;
  pointer-events: none !important;
}`

**Comment by LaurenceLong** [comment]
If you are using embed sdk, I suggest the css below to separate superset itself and embed dashboard

Edit Dashboard > Edit CSS

body:not(:has(.navbar-brand)) #GRID_ID-panel-GRID_ID * div.header-title > span {
    pointer-events: none !important;
    cursor: default !important;
    text-decoration: none !important;
    color: inherit !important;
}

**Comment by SuperMAB** [comment]
After hours of reflexion 
The embeding SDK seems to grant "Gamma" Role automatically even if you force it to another role
So the solution was to edit permissions of Gamma Role.
Here is  an example of the minimum you can set : 
<img width="1204" height="369" alt="image" src="https://github.com/user-attachments/assets/40d7cbd8-75fe-482d-a7e6-6824cb1fca1d" />
1- all datasource access on all_datasource_access : for all sources ( you can resctrict if you want)
2- can csv on Superset : to export CSV, EXCEL and image
3- can explore json on Superset : needed for some charts like map and heatmap because export excel is generating JSON (I assume it's another issue)

**Reply by marianysilva** [reply]
Hey @SuperMAB you need to setup the [GUEST_ROLE_NAME](https://github.com/apache/superset/blob/master/superset/config.py#L2054) in your config.py file, so you can use a custom role.
